src/app/material-list/material-list.component.ts

```diff
@@ -0,0 +1,74 @@
+import { Component } from '@angular/core';
+import { MaterialService } from '../material.service';
+
+@Component({
+  selector: 'app-material-list',
+  templateUrl: './material-list.component.html',
+  styleUrls: ['./material-list.component.css']
+})
+export class MaterialListComponent {
+
+  public materials = [];
+
+  selectedMaterial;
+
+  constructor(private _materialService: MaterialService) {
+    this.getMaterials();
```

Comment:
API calls should probably use lifecycle hooks like `ngOnInit`.
https://angular.io/guide/lifecycle-hooks